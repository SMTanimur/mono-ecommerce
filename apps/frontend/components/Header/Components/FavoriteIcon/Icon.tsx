import React  from "react";
import clsx from "clsx";
import { NavButton } from "@ecommerce/shared/ui/NavButton";

interface Props {
  favoriteItemsQuantity: number;
}

const Icon: React.FC<Props> = ({favoriteItemsQuantity}) => {
  return (
    <NavButton type="heart">
      {favoriteItemsQuantity > 0 && (
        <span
          className={clsx(
            'h-4.75 top-0 right-0 w-4.75 absolute',
            'flex items-center justify-center',
            'bg-sky-600',
            'rounded-full',
            'text-xs text-white'
          )}
          aria-hidden="true"
        >
          {favoriteItemsQuantity}
        </span>
      )}
    </NavButton>
  );
};

export default Icon;
