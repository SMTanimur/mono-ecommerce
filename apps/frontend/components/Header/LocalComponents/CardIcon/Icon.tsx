import React from 'react';
import clsx from 'clsx';
import { NavButton } from '@ecommerce/shared/ui/NavButton';

interface Props {
  cartItemsQuantity: number;
}

const Icon: React.FC<Props> = ({ cartItemsQuantity }) => {
  return (
    <NavButton type="cartBlack">
      {cartItemsQuantity > 0 && (
        <span
          className={clsx(
            'h-4.75 top-0 right-0 w-4.75 absolute',
            'flex items-center justify-center',
            'bg-sky-600',
            'rounded-full',
            'text-xs text-white'
          )}
          aria-hidden="true"
        >
          {cartItemsQuantity}
        </span>
      )}
    </NavButton>
  );
};

export default Icon;
