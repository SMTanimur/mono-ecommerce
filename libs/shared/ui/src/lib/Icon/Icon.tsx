import FeatherIcon from 'feather-icons-react';
import clsx from 'clsx';
import { Spinner } from './Components/Spinner';
import {SVGAttributes} from "react";

export interface IconProps extends SVGAttributes<SVGElement> {
  icon?: keyof typeof ICONS;
  loading?: boolean;
  classes?: string;
  size?: typeof SIZES[number];
}

export const ICONS = {
  cart: 'shopping-cart',
  cartBlack: 'shopping-cart',
  heartFilled: 'heart',
  user: 'user',
  arrowRight: 'arrow-right',
  arrowRightBlack: 'arrow-right',
  'arrow-right': 'arrow-right',
  heart: 'heart'
} as const;

const SIZES = ['sm', 'base', 'lg'] as const;

export const Icon: React.FC<IconProps> = (props: IconProps) => {
  const { icon = 'cart', classes, size = 'base', loading, ...other } = props;

  return (
    <div
      className={clsx(
        'relative',
        'flex items-center',
        'select-none pointer-events-none',

        size === 'sm' && 'h-4 w-4  sm:(h-6 w-6)',
        size === 'base' && 'h-5 w-5',
        size === 'lg' && 'h-8 w-8',

        classes
      )}
      aria-hidden="true"
    >
      {loading ? (
        <Spinner />
      ) : (
        <FeatherIcon icon={ICONS[icon] || icon} {...other} />
      )}
    </div>
  );
};
