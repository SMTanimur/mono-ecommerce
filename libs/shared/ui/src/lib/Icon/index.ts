import { Icon, IconProps, ICONS } from './Icon';

export { Icon, ICONS };

export type { IconProps };
