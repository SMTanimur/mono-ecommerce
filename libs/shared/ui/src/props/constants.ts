import { ButtonProps } from '@ecommerce/shared/ui/Button';
import {CartItemProps} from "@ecommerce/shared/ui/CartItem";
import {ProductSidebarProps} from "@ecommerce/shared/ui/ProductSidebar";
import {CartSidebarProps} from "@ecommerce/shared/ui/CartSidebar";
import {CategoryCardProps} from "@ecommerce/shared/ui/CategoryCard";
import {IconProps} from "@ecommerce/shared/ui/Icon";
import {ItemCardProps} from "@ecommerce/shared/ui/ItemCard";
import {NavButtonProps} from "@ecommerce/shared/ui/NavButton";
import {ProductGalleryProps} from "@ecommerce/shared/ui/ProductGallery";

import cat from './cat.jpg';
import cat2 from './cat2.jpeg';

console.log(cat2)

export const PRODUCT: ProductSidebarProps['product'] = {
  slug: 'some-name',
  price: 65,
  name: 'Some name',
  images: {
    data: [
      {
        attributes: {
          url: cat,
        },
      },
      {
        attributes: {
          url: cat2,
        },
      },
    ],
  },
  smallDescription: 'Small Description',
  description: 'Description',
  information: [
    {
      title: 'Title1',
      content: 'content1',
    },
  ],
};

export const ITEM: ItemCardProps['item'] = {
  slug: 'cat-cat',
  price: 65,
  name: 'cat cat',
  images: {
    data: [
      {
        attributes: {
          url: cat,
        },
      },
      {
        attributes: {
          url: cat2,
        },
      },
    ],
  },
  smallDescription: 'smallDescription',
  description: 'description',
};

export const BUTTON_DEFAULT_PROPS: ButtonProps = {
  children: 'hello',
  tag: 'button',
  size: 'base',
  type: 'primary',
  round: 'full',
};

export const CART_ITEM_DEFAULT_PROPS: CartItemProps = {
  inCartAmount: 1,
  item: PRODUCT,
  handleCartMinusClick: () => {
    console.log('cart minus');
  },
  handleCartPlusClick: () => {
    console.log('cart plus');
  },
  handleRemoveFromCart: () => {
    console.log('cart remove');
  },
  toggleFavorite: () => {
    console.log('favorite');
  },
};

export const CART_SIDEBAR_DEFAULT_PROPS: CartSidebarProps = {
  cartTotal: 100,
  isUser: true,
  handleCheckoutClick: async () => {
    console.log('checkout');
  },
  isLoading: false,
};

export const CATEGORY_CARD_DEFAULT_PROPS: CategoryCardProps = {
  category: {
    smallDescription:
      'smalll des',
    image: {
      data: {
        attributes: {
          url: cat,
        },
      },
    },
    slug: '1',
  },
};
export const ICON_DEFAULT_PROPS: IconProps = {
  icon: 'cart',
};

export const ITEM_CARD_DEFAULT_PROPS: ItemCardProps = {
  item: ITEM,
  size: 'base',
  handleCartClick: () => console.log('cart clicked'),
  handleAmountCartClick: () => console.log('amount cart clicked'),
  handleFavoriteClick: () => console.log('favorite clicked'),
};
export const NAV_BUTTON_DEFAULT_PROPS: NavButtonProps = {
  type: 'heart',
};
export const PRODUCT_GALLERY_DEFAULT_PROPS: ProductGalleryProps = {
  images: [
    {
      attributes: {
        url: cat,
      },
    },
    {
      attributes: {
        url: cat2,
      },
    },
  ],
};
