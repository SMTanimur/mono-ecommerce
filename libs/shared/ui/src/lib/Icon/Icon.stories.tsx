import { ICON_DEFAULT_PROPS } from '@ecommerce/shared/props/constants';
import { Story, Meta } from '@storybook/react';
import { Icon, IconProps } from './Icon';

export default {
  component: Icon,
  title: 'Components/Icon',
} as Meta;

const Template: Story<IconProps> = (args) => <Icon {...args} />;

export const Primary = Template.bind({});
Primary.args = ICON_DEFAULT_PROPS;
