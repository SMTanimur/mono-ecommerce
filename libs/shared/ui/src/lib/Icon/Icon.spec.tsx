import { ICON_DEFAULT_PROPS } from '../../props/constants';
import { render, RenderResult, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import '@testing-library/jest-dom';

import { Icon } from './Icon';

describe('Icon', () => {
  let renderResult: RenderResult;
  const props = ICON_DEFAULT_PROPS;

  beforeEach(() => {
    renderResult = render(<Icon {...props} />);
  });

  it('should render successfully', () => {
    const { baseElement } = renderResult;

    const tree = create(<Icon {...props} />).toJSON();
    expect(tree).toMatchSnapshot();

    expect(baseElement).toBeTruthy();
  });

  it('when loading show spinner instead of icon', () => {
    expect(screen.queryByTestId('spinner')).not.toBeInTheDocument();

    renderResult.rerender(<Icon {...props} loading={true} />);

    expect(screen.queryByTestId('spinner')).toBeInTheDocument();
  });
});
