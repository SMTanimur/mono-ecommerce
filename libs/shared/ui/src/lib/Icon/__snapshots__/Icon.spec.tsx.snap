// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render successfully 1`] = `
<div
  aria-hidden="true"
  className="relative flex items-center select-none pointer-events-none h-5 w-5"
>
  <svg
    className="feather feather-shopping-cart "
    fill="none"
    height={24}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    viewBox="0 0 24 24"
    width={24}
  >
    <g
      dangerouslySetInnerHTML={
        Object {
          "__html": "<circle cx=\\"9\\" cy=\\"21\\" r=\\"1\\"></circle><circle cx=\\"20\\" cy=\\"21\\" r=\\"1\\"></circle><path d=\\"M1 1h4l2.68 13.39a2 2 0 0 0 2 1.61h9.72a2 2 0 0 0 2-1.61L23 6H6\\"></path>",
        }
      }
    />
  </svg>
</div>
`;
